packages/vm/src/parsed_wasm.rs

```diff
@@ -0,0 +1,70 @@
+use wasmer::wasmparser::{
+    Export, Import, MemoryType, Parser, TableType, ValidPayload, Validator, WasmFeatures,
+};
+
+use crate::VmResult;
+
+/// A parsed and validated wasm module.
+/// It keeps track of the parts that are important for our static analysis and compatibility checks.
+#[derive(Debug)]
+pub struct ParsedWasm<'a> {
```

Comment:
Do we need this lifetime `'a`? Seems like it saves a few copies of import/export names but that's it. Or do we need it in order to be able to reuse the Import and Export types?

Reply:
Yes, we need it to reuse the types